tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -109,61 +210,122 @@ spec:
             fi
 
             for manifest_digest in $manifest_digests; do
-              for tag in ${TAGS}; do
-                # Iterate over both rh-registry-repo and registry-access-repo
-                for registry_reference in ${rh_registry_repo} ${registry_access_repo}; do
-                  echo "Creating InternalRequest to sign image with tag ${tag}:"
-                  echo "- reference=${registry_reference}:${tag}"
-                  echo "- manifest_digest=${manifest_digest}"
-                  echo "- requester=$(params.requester)"
-
-                  internal-request -r "${request}" \
-                      -p pipeline_image=${pipeline_image} \
-                      -p reference=${registry_reference}:${tag} \
-                      -p manifest_digest=${manifest_digest} \
-                      -p requester=$(params.requester) \
-                      -p config_map_name=${config_map_name} \
-                      -l ${TASK_LABEL}=${TASK_ID} \
-                      -l ${PIPELINERUN_LABEL}=$(params.pipelineRunUid) \
-                      -s false
-                  ((++count))
-
-                  if [ "$count" -eq "$N" ]; then
-                      wait-for-ir -l ${TASK_LABEL}=${TASK_ID} -t $(params.requestTimeout)
-                      count=0
+
+              set -x
+              find_signatures --pyxis-graphql-api "${PYXIS_GRAPHQL_URL}" \
+                  --manifest_digest "${manifest_digest}" \
+                  --repository "${repository}" \
+                  --output_file "/tmp/${manifest_digest}"
+              set +x
+
+              # Iterate over both rh-registry-repo and registry-access-repo
+              for registry_reference in ${rh_registry_repo} ${registry_access_repo}; do
+
+                for tag in ${TAGS}; do
+
+                  # is tag signed?
+                  set +e
+                  grep -q "^${registry_reference}:${tag}$" "/tmp/${manifest_digest}"
+                  notFound=$?
+                  set -e
+
+                  if [ "$notFound" == 1 ] ; then
+                    while (( ${RUNNING_JOBS@P} >= "$CONCURRENT_LIMIT" )); do
+                      wait -n
+                    done
+
+                    echo "Creating ${requestType} to sign image with tag ${tag}:"
+                    echo "- reference=${registry_reference}:${tag}"
+                    echo "- manifest_digest=${manifest_digest}"
+                    echo "- requester=$(params.requester)"
+
+                    ${requestType} -r "${request}" \
+                      -p pipeline_image="${pipeline_image}" \
+                      -p reference="${registry_reference}:${tag}" \
+                      -p manifest_digest="${manifest_digest}" \
+                      -p requester="$(params.requester)" \
+                      -p config_map_name="${config_map_name}" \
+                      -p taskGitUrl="$(params.taskGitUrl)" \
```

Comment:
For internal-pipeline you have both `-p taskGitUrl=` and `--task-git-url`? Looking at the script: https://github.com/konflux-ci/release-service-utils/blob/main/utils/internal-pipelinerun
One is passed as pipeline param, the other is used in pipeline ref. So it's kind of redundant.

But I guess it would just be a bit more hassle to use just one, right? If you used -p only, then in the script you would need to find it in the list of params to use it in the pipelineref. So probably better would be to only use --task-git-url and add that also as an extra param in the scripts, but this would require changes in both scripts. Am I getting this right? (Same with the other git arg.) So I guess it's fine for now and we can improve it later if we want to.

Reply:
Actually, for IRs even the `-p` git parameters are not used, right? So yeah, it would be better to only have the `--task-git*` args here and then in `internal-pipeline` you would add them to the pipeline params as well. But it can be done later. Let me know and I can create a Jira for it.

Reply:
I see your point. I'd prefer to defer this to another issue.

Reply:
Created jira: https://issues.redhat.com/browse/RELEASE-1233